src/constants/validation.constants.ts

```diff
@@ -0,0 +1,187 @@
+     * @example ValidationConstants.patterns.tradingPlatformInvestorPassword.test("Password1!$")
+     **/
+    tradingPlatformInvestorPassword:
+        /^(?=.*[a-z])(?=.*[0-9])(?=.*[A-Z])(?=.*[!@#$%^&*()+\-=\[\]{};':\"|,\.<>\?_~])[ -~]{8,16}$/,
+    /**
+     * @regex /^[A-Za-z]+([a-zA-Z.' -])*[a-zA-Z.' -]+$/
+     * @description This pattern matches any string that starts with one or more letters (uppercase or lowercase), followed by zero or more occurrences of letters, and any of these characters (.' -) and ends with one or more occurrences of letters, and any of these characters (.' -).
+     * @example ValidationConstants.patterns.letterSymbols.test("John Doe")
+     * @example ValidationConstants.patterns.letterSymbols.test("John-Doe")
+     * @example ValidationConstants.patterns.letterSymbols.test("John O'Doe")
+     */
+    letterSymbols: /^[A-Za-z]+([a-zA-Z.' -])*[a-zA-Z.' -]+$/,
+    /**
+     * @regex /^(?!.*\s{2,})[\p{L}\s'.-]{2,50}$/u
+     * @description This pattern matches any string that contains 2-50 characters, starts and ends with valid characters (letters, whitespace, period, single quote, or hyphen).
+     * @example ValidationConstants.patterns.name.test("John Doe")
+     * @example ValidationConstants.patterns.name.test("John-Doe")
+     * @example ValidationConstants.patterns.name.test("John O'Doe")
+     * @example ValidationConstants.patterns.name.test("John O. Doe")
+     */
+    name: /^(?!.*\s{2,})[\p{L}\s'.-]{2,50}$/u,
+    /**
+     * @regex /[`~!@#$%^&*)(_=+[}{\]\\/";:?><|]+/
+     * @description This pattern matches any string that contains any of these characters: `~!@#$%^&*)(_=+[}{\]\\/";:?><|
+     * @example ValidationConstants.patterns.general.test("Password1!")
+     */
+    general: /[`~!@#$%^&*)(_=+[}{\]\\/";:?><|]+/,
+    /**
+     * @regex /[a-z]/
+     * @description This pattern matches any string that contains lowercase letters.
+     * @example ValidationConstants.patterns.lowercase.test("abc")
+     */
+    lowercase: /[a-z]/,
+    /**
+     * @regex /\d/
+     * @description This pattern matches any string that contains digits.
+     * @example ValidationConstants.patterns.number.test("pets123")
+     */
+    number: /\d/,
+    /**
+     * @regex /\W/
+     * @description This pattern matches any string that contains special characters.
+     * @example ValidationConstants.patterns.specialCharacter.test("Password1!")
+     */
+    specialCharacter: /\W/,
+    /**
+     * @regex /[A-Z]/
+     * @description This pattern matches any string that contains uppercase letters.
+     * @example ValidationConstants.patterns.uppercase.test("ABC")
+     */
+    uppercase: /[A-Z]/,
+    /**
+     * @regex /^.{8,25}$/
+     * @description This pattern matches any string that contains 8-25 characters.
+     * @example ValidationConstants.patterns.between8and25Characters.test("Password1!")
+     */
+    between8and25Characters: /^.{8,25}$/,
```

Comment:
hi @jim-deriv , I think we need a check for length between 8-16 which MT5 new Password Policy follows

for non-MT5 CFDs (dxtrade, cTrader) will still be needing length 8-25 so can keep this

Reply:
Hi @shontzu-deriv, Alright, I'll add it as well

Reply:
thankies! aside from that all lgtm